### ISSUE TITLE
Filtering of DataFrame based on the column with dates in Date32 format is not working

### ISSUE THREAD

**Issue by solisortusaeternum** [pola-rs/polars#1332]
#### Are you using Python or Rust?

**Python**

#### What version of polars are you using?

polars==0.9.4

#### What operating system are you using polars on?

Ubuntu 20.04.1 LTS

#### Describe your bug.

Polars DataFrame is returning wrong results when comparison operation is performed on dates using pl.lit_date().

#### What are the steps to reproduce the behavior?

```
import polars as pl
from datetime import datetime

# create a simple dataset
dataset = pl.DataFrame({"date": ["2020-01-02", "2020-01-03", "2020-01-04"], "index": [1, 2, 3]})
df = dataset.with_column(pl.col("date").str.strptime(pl.datatypes.Date32, "%Y-%m-%d"))

# filter out the data to match only records from the previous year
df.filter(col('date') <= pl.lit_date(datetime(2019, 1, 3)))
```

#### What is the actual behavior?

```
df.filter(col('date') <= pl.lit_date(datetime(2019, 1, 3)))

shape: (3, 2)
┌──────────────┬───────┐
│ date         ┆ index │
│ ---          ┆ ---   │
│ date32(days) ┆ i64   │
╞══════════════╪═══════╡
│ 2020-01-02   ┆ 1     │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┤
│ 2020-01-03   ┆ 2     │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┤
│ 2020-01-04   ┆ 3     │
└──────────────┴───────┘
```

#### What is the expected behavior?

Empty DataFrame should be returned as the result.

**Comment by ritchie46** [comment]
Thanks for the issue repoort. On it!

**Comment by ritchie46** [comment]
This snippet works on my machine, this likely is due to timezones. Does anyone know how to get a `UTC` timestamp from a python datetime object?

**Comment by solisortusaeternum** [comment]
> This snippet works on my machine, this likely is due to timezones. Does anyone know how to get a `UTC` timestamp from a python datetime object?

yep, I already tried to add timezone, didn't help :) but if Date64 format is used everything works fine

**Comment by ritchie46** [comment]
> yep, I already tried to use UTC timestamps and to convert timestamp back to datetime, didn't help :) but if Date64 format is used everything works fine

Right, thank, now I now what the issue is. :)

**Comment by ritchie46** [comment]
Should be fixed in #1335

**Comment by solisortusaeternum** [comment]
> Should be fixed in #1335

thanks a lot!

**Comment by ritchie46** [comment]
Fixed and being released at the moment: https://github.com/pola-rs/polars/pull/1339/checks?check_run_id=3569701202
